ios/RNSoundPlayer.m

```diff
@@ -89,6 +89,17 @@ @implementation RNSoundPlayer
     [session setActive:true error:nil];
 }
 
+RCT_EXPORT_METHOD(setMixAudio:(BOOL) on) {
```

Comment:
One caveat is that if a user uses `setSpeaker` with this option, it could produce unexpected outcomes. For example, in it's current implementation if you set `setSpeaker(true)`, it sets the category to `AVAudioSessionCategoryPlayAndRecord`. But if you then used `setMixAudio`, it would revert the category back to the default `AVAudioSessionCategoryPlayback`. 

I figure a better option might be a function to combine them, e.g. `setPlaybackSettings(overrideSpeaker = false, mixAudio = false)`, but this was my first time in Objective C so decided to keep it simple :) Flagging in case you want to improve on it.